test/unit/app/controllers/permissions/permissions-controller-test.js

```diff
@@ -461,6 +461,93 @@ describe('permissions controller', function () {
     })
   })
 
+  describe('addPermittedAccount', function () {
+    let permController, notifications
+
+    beforeEach(function () {
+      notifications = initNotifications()
+      permController = initPermController(notifications)
+      grantPermissions(
+        permController, ORIGINS.a,
+        PERMS.finalizedRequests.eth_accounts(ACCOUNT_ARRAYS.a)
+      )
+      grantPermissions(
+        permController, ORIGINS.b,
+        PERMS.finalizedRequests.eth_accounts(ACCOUNT_ARRAYS.b)
+      )
+    })
+
+    it('should throw if account is not a string', async function () {
+      await assert.rejects(
+        () => permController.addPermittedAccount(ORIGINS.a, {}),
+        ERRORS.validatePermittedAccounts.nonKeyringAccount({}),
+        'should throw on non-string account param'
+      )
+    })
+
+    it('should throw if given account isn\'t in keyring', async function () {
```

Comment:
nit: I'm (perhaps irrationally) bothered by the escaped single quote
```suggestion
    it('should throw if given account is not in keyring', async function () {
```